src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,94 @@
+package duke;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.Scanner;
+
+import duke.commands.Command;
+import duke.ui.CliUi;
+import duke.ui.JavafxUi;
+import duke.ui.Ui;
+
+/**
+ * Main Duke class.
+ */
+public class Duke {
+    private Ui ui;
+    private Storage storage;
+    private TaskList tasks;
+
+    public Duke(boolean isCli) {
+        this(null, isCli);
+    }
+
+    public Duke(String directoryPath, boolean isCli) {
+        if (isCli) {
+            ui = new CliUi();
+        } else {
+            ui = new JavafxUi();
+        }
+        if (directoryPath != null) {
+            storage = new Storage(Paths.get(directoryPath));
+        } else {
+            storage = new Storage();
+        }
+        try {
+            tasks = storage.readTasks();
+        } catch (IOException e) {
+            ui.printError("Unable to create file");
+            tasks = new TaskList();
+        } catch (Exception e) {
+            ui.printError("Unable to parse file");
+            tasks = new TaskList();
+        }
+    }
+
+
+    /**
+     * Executes the Duke program by reading in commands and giving appropriate outputs.
+     */
+    public void run() {
+        ui.printIntro();
+
+        Scanner stdin = new Scanner(System.in);
+        String line = stdin.nextLine();
+        boolean end = false;
```

Comment:
should boolean variables be named to sound like booleans?

Reply:
renamed to `isExit`